external-api/api-tests/v1/party-db.api-test.ts

```diff
@@ -12,19 +12,81 @@
 
 import MockAdapter from 'axios-mock-adapter';
 import axios, { HttpStatusCode } from 'axios';
-import { MOCK_COMPANY_REGISTRATION_NUMBERS } from '@ukef/dtfs2-common';
+import { MOCK_COMPANY_REGISTRATION_NUMBERS, isAutomaticSalesforceCustomerCreationFeatureFlagEnabled } from '@ukef/dtfs2-common';
 import { app } from '../../src/createApp';
 import { api } from '../api';
 
 const { APIM_MDM_URL } = process.env;
 const { VALID, VALID_WITH_LETTERS } = MOCK_COMPANY_REGISTRATION_NUMBERS;
 const { get } = api(app);
+let axiosMock: MockAdapter;
 
-const axiosMock = new MockAdapter(axios);
-axiosMock.onGet(`${APIM_MDM_URL}customers?companyReg=${VALID}`).reply(HttpStatusCode.Ok, {});
-axiosMock.onGet(`${APIM_MDM_URL}customers?companyReg=${VALID_WITH_LETTERS}`).reply(HttpStatusCode.Ok, {});
+jest.mock('@ukef/dtfs2-common', () => ({
+  ...jest.requireActual('@ukef/dtfs2-common'),
+  isAutomaticSalesforceCustomerCreationFeatureFlagEnabled: jest.fn(),
+}));
+
+beforeEach(() => {
+  axiosMock = new MockAdapter(axios);
+
+  axiosMock.onGet(`${APIM_MDM_URL}customers?companyReg=${VALID}`).reply(HttpStatusCode.Ok, {});
+  axiosMock.onGet(`${APIM_MDM_URL}customers?companyReg=${VALID_WITH_LETTERS}`).reply(HttpStatusCode.Ok, {});
+  axiosMock.onGet(`${APIM_MDM_URL}customers/salesforce?companyRegistrationNumber=${VALID}`).reply(HttpStatusCode.Ok, {});
+  axiosMock.onGet(`${APIM_MDM_URL}customers/salesforce?companyRegistrationNumber=${VALID_WITH_LETTERS}`).reply(HttpStatusCode.Ok, {});
+});
+
+afterEach(() => {
+  axiosMock.resetHistory();
+});
+
+describe('when automatic Salesforce customer creation feature flag is disabled', () => {
+  beforeEach(() => {
+    jest.mocked(isAutomaticSalesforceCustomerCreationFeatureFlagEnabled).mockReturnValue(false);
+  });
+
+  describe('/party-db', () => {
+    describe('GET /party-db', () => {
+      it(`returns a ${HttpStatusCode.Ok} response with a valid companies house number`, async () => {
+        const { status } = await get(`/party-db/${VALID}`);
+
+        expect(status).toEqual(HttpStatusCode.Ok);
+      });
+
+      it(`returns a ${HttpStatusCode.Ok} response with a valid companies house number`, async () => {
+        const { status } = await get(`/party-db/${VALID_WITH_LETTERS}`);
+
+        expect(status).toEqual(HttpStatusCode.Ok);
+      });
+
+      it(`calls the correct url`, async () => {
+        await get(`/party-db/${VALID}`);
+
+        expect(axiosMock.history.get).toHaveLength(1);
```

Comment:
We don't really use `axiosMock.history` in the DTFS codebase, I would suggest assert using `toHaveBeenCalledWith...` with `spy` please.